src/nested_pandas/utils/utils.py

```diff
@@ -0,0 +1,38 @@
+import pandas as pd
+
+from nested_pandas import NestedFrame
+
+
+def count_nested(df, nested, by=None, join=True) -> NestedFrame:
+    """Counts the number of rows of a nested dataframe.
+
+    Parameters
+    ----------
+    df: NestedFrame
+        A NestedFrame that contains the desired `nested` series
+        to count.
+    nested: 'str'
+        The label of the nested series to count.
+    by: 'str', optional
+        Specifies a column within nested to count by, returning
+        a count for each unique value in `by`.
+    join: bool, optional
+        Join the output count columns to df and return df, otherwise
+        just return a NestedFrame containing only the count columns.
+
+    Returns
+    -------
+    NestedFrame
+    """
+
+    if by is None:
+        counts = df["nested"].apply(lambda x: len(x)).rename(f"n_{nested}")
+    else:
+        counts = df["nested"].apply(lambda x: x[by].value_counts())
+        counts = counts.rename(columns={colname: f"n_{nested}_{colname}" for colname in counts.columns})
+    if join:
+        return df.join(counts)
+    # else just return the counts NestedFrame
+    if isinstance(counts, pd.Series):
+        counts = NestedFrame(counts.to_frame())
```

Comment:
Could you please explain when do we get here? Is it in case of `by is None`? May we do this transformation their instead?

Reply:
Yeah, when `by is None` we just return a single series as it just returns a single len value for each nested dataframe. There's a small advantage to doing it here, in that when `Join=True` the join accepts either a series or dataframe, so we avoid having to cast it to a dataframe for the `Join=True` case. The overhead is probably very minimal though

Reply:
Thanks! Could you please add a short comment, like `# for by=None`, or move it to that `if`-block?

Reply:
Added a comment!